website/pages/blog/setting-up-graphql-server-to-scale-with-graphql-codegen-server-preset.mdx

```diff
@@ -0,0 +1,440 @@
+
+Using the same example before, we can identify 3 main domains: User, Auth and Book, each have CODEOWNERS set up to notify appropriate teams of changes. The Maintainers own `server.ts` and other configs.
+
+```markdown
+├── src/
+│   ├── schema/
+│   │   ├── user/                      # Team A notified if changed
+│   │   │   ├── datasources.ts
+│   │   │   ├── resolvers.ts
+│   │   │   ├── schema.graphql
+│   │   ├── auth/                      # Team A notified if changed
+│   │   │   ├── datasources.ts
+│   │   │   ├── resolvers.ts
+│   │   │   ├── schema.graphql
+│   │   ├── book/                      # Team B notified if changed
+│   │   │   ├── datasources.ts
+│   │   │   ├── resolvers.ts
+│   │   │   ├── schema.graphql
+│   ├── server.ts                      # Maintainers notified if changed
+│   ├── codegen.yml                    # Maintainers notified if changed
+```
+
+## The Hard Problems
+
+Splitting schema into modules is usually easy to get teams to agree on. It is then we start to see the hard problems: how to enforce best practices to the teams while reducing the time Maintainers need to spend on server maintenance.
+
+### 1. How to Enforce Best Practices for All Teams
+
+Bad practices and conventions spread like fire, on a hot summer day, in the bush, in Australia. Trust me it is the worst! I used to be part of a Maintainers team. We had guidelines various topics, one of them is resolver naming convention. Once in a blue moon, one person incorrectly used pascal case instead of camel case. The following day, I woke up with more than half of the resolvers in pascal case. 😱
+
+OK, I am exaggerating here but the experience was very traumatising.
+
+Guidelines alone is only good if people follow it. Without explicit and automatic enforcement, things start to slip and bad practices spread sooner or later. 
+
+To enforce guidelines, we need tools. Luckily, these days we have an extensive range of tools for best practices a GraphQL server:
+
+*   [GraphQL Codegen](https://the-guild.dev/graphql/codegen) with [typescript](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript) and [typescript-resolvers](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-resolvers) plugins for type-safe GraphQL server development in TypeScript
+*   [GraphQL ESLint](https://github.com/B2o5T/graphql-eslint) for validation and conventions like naming and best practices.
+
+However, there are still areas to improve.
+
+For example, the guideline for one of the GraphQL servers I am working on is to use generated types from GraphQL Codegen. However, some people do not follow the guidelines, mostly because they are not aware of it. Luckily, other team members or the Maintainers catch these issues at PR review time. But it could have saved everyone time and effort if we had tools that could enforce this guideline.
```

Comment:
**Thought:** Does the below sentences sound too harsh to both the existing tooling and teams I'm working with?